internals/overlord/overlord.go

```diff
@@ -49,6 +49,25 @@ var (
 	defaultCachedDownloads = 5
 )
 
+// Extension represents an extension of the Overlord.
+type Extension interface {
+	// ExtraManagers is called when building an Overlord, it allows additional
+	// managers to be added to the Overlord.
+	ExtraManagers(o *Overlord) ([]StateManager, error)
+}
+
+// Options is the arguments passed to construct an Overlord.
+type Options struct {
+	// PebbleDir is the path to the pebble directory. It must be provided.
+	PebbleDir string
+	// RestartHandler is an optional structure to handle restart requests.
+	RestartHandler restart.Handler
+	// ServiceOutput is an optional output for the logging manager.
+	ServiceOutput io.Writer
+	// Extension, if set, defines additional components to add to the overlord.
```

Comment:
```
// Extension allows extending the overlord with externally defined features.
```

This is what the extension is. The documentation at the moment is describing ExtraManagers, but that's up to the Extension itself to document.